src/ucp/proto/proto_perf.c

```diff
@@ -566,6 +506,10 @@ ucs_status_t ucp_proto_perf_sum(const ucp_proto_perf_t *perf,
 
     *flat_perf_ptr = flat_perf;
     return UCS_OK;
+
+err_cleanup:
+    ucp_proto_flat_perf_destroy(flat_perf);
```

Comment:
maybe add helper function flat_perf_alloc?

Reply:
Do you mean something like:

```c
static ucs_status_t ucp_proto_flat_perf_alloc(ucp_proto_flat_perf_t **flat_perf_ptr)
{
    *flat_perf_ptr = ucs_malloc(sizeof(**flat_perf_ptr), "flat perf");
    return (*flat_perf_ptr != NULL) ? UCS_OK : UCS_ERR_NO_MEMORY;
}
```

Reply:
something like
```c
static ucs_status_t ucp_proto_flat_perf_alloc(ucp_proto_flat_perf_t **flat_perf_p)
{
    ucp_proto_flat_perf_t *flat_perf;

    flat_perf = ucs_malloc(sizeof(*flat_perf_ptr), "flat_perf");
    if (flat_perf == NULL) {
        return UCS_ERR_NO_MEMORY;
    }
    
    *flat_perf_p = flat_perf;
    return UCS_OK;
}

Reply:
Done